classes/kernel-module-signing.bbclass

```diff
@@ -7,19 +7,20 @@ SIGN_FILE = "${B}/scripts/sign-file"
 export KERNEL_MODULE_SIG_CERT
 
 do_configure_append() {
-    if [ -n "${KERNEL_MODULE_SIG_CERT}" ] &&
-       grep -q '^CONFIG_MODULE_SIG=y' ${B}/.config ; then
-        sed -i -e '/CONFIG_MODULE_SIG_KEY[ =]/d' ${B}/.config
-        echo "CONFIG_MODULE_SIG_KEY=\"${KERNEL_MODULE_SIG_CERT}\"" >> \
-               ${B}/.config
-        sed -i -e '/CONFIG_MODULE_SIG_ALL[ =]/d' ${B}/.config
-        echo "# CONFIG_MODULE_SIG_ALL is not set" >> \
-               ${B}/.config
+    if ! grep -q '^CONFIG_MODULE_SIG=y' ${B}/.config ; then
+        return
+    fi
+    if [ -z "${KERNEL_MODULE_SIG_CERT}" ]; then
+        bbfatal "Kernel module signing should only be used when setting \
```

Comment:
Something to note here.  The bbfatal message is printed and then bitbake prints the contents of log.do_configure, which is quite long with all the `make oldconfig` output.  All you see at the end of it is: 

```
make[1]: Entering directory '.../build'
  GEN     Makefile
# 
# No change to .config
# 
make[1]: Leaving directory '.../build'
make: Leaving directory '/home/build/openxt/build-201218/tmp-glibc/work-shared/usbvm/kernel-source'
WARNING: .../temp/run.do_configure.7382:1 exit 1 from 'exit 1'

ERROR: Logfile of failure stored in: .../temp/log.do_configure.7382
```

If you look back far enough you see the message.  It's on line 2299 of the 4596 line log.do_configure.  Not a blocker, but something I noticed.

Reply:
We could just `bberror` and `exit 1` I guess.

Reply:
That didn't change the output :/